common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -279,6 +279,69 @@ def _is_in_bulk_operation(self, course_key, ignore_case=False):
         return self._get_bulk_ops_record(course_key, ignore_case).active
 
 
+class UnsetValueError(Exception):
+    """
+    Throw this when accessing an unset value.
+    """
+    pass
+
+
+class UnsetValue(object):
+    """
+    Simple class which supports an "unset" value as well as None and non-None values.
+    """
+    def __init__(self, value=None, valid=False):
+        self._value = value
+        self.valid = valid
+
+    @property
+    def value(self):
+        """
+        Getter for the value. Raises UnsetValueError if value is unset.
+        """
+        if self.valid:
+            return self._value
+        else:
+            raise UnsetValueError()
+
+    @value.setter
+    def value(self, value):
+        """
+        Setter for the value. Also marks the value as valid.
+        """
+        self._value = value
+        self.valid = True
```

Comment:
A pattern we've used for this in other places is somewhat simpler:

    UNSET = object()  # This guarantees that it UNSET == UNSET, but doesn't == anything else

And then you can check `is UNSET` to find out if a value is unset or not.